gulpfile.js

```diff
@@ -376,6 +378,50 @@ export function restartServer(done) {
   }
 }
 
+export function runGithubChangelogGenerator() {
+  return execa.command('github_changelog_generator -u educandu -p educandu --since-tag 0.1.0', { stdio: 'inherit' });
+}
+
+export async function generateJiraLinks() {
+  let fileStream = null;
+  const newLines = [];
+  try {
+    fileStream = createReadStream('CHANGELOG.md');
+    const lineReader = readline.createInterface({
+      input: fileStream,
+      crlfDelay: Infinity
+    });
+
+    for await (const line of lineReader) {
+      let jiraLine = null;
+      if ((/^- EDU-\d+/).test(line)) {
+        const jiraNumber = line.match(/EDU-\d+/);
+        jiraLine = line.replace(/^- EDU-\d+/, `- [${jiraNumber}](https://educandu.atlassian.net/browse/${jiraNumber})`);
+      }
+
+      newLines.push(jiraLine || line);
+    }
+    lineReader.close();
+    fileStream.close();
+  } finally {
+    fileStream && fileStream.close();
+  }
+
+  let writer = null;
+  try {
+    writer = createWriteStream('CHANGELOG.md');
+    for (const line of newLines) {
+      writer.write(line);
+      writer.write(EOL);
+    }
+    writer.end();
+  } finally {
+    writer && writer.close();
+  }
+}
```

Comment:
Just out of curiosity, why not simply `fs.readFile` and `fs.writeFile`, the whole function would then just be:

~~~
export async function generateJiraLinks() {
  const changeLog = await fs.readFile('CHANGELOG.md', 'utf8');
  const replacedChangeLog = changeLog.replace(/EDU-\d+/g, num => `[${num}] 
 (https://educandu.atlassian.net/browse/${num})`);
}
await fs.writeFile('CHANGELOG.md', replacedChangeLog, 'utf8');
~~~

Reply:
Several reasons:
 1) I was doing the first thing that comes to mind to try it out.
 2) I didn't work with the file system in node up to now, and I wanted to process each line since it seems like it replaces the whole file (at least that is the behavior when we don't have tags). This would mean that the second time you run your script this would replace the EDU-d everywhere again, which would lead to broken links. If you look at my regex, it looks for exact matching start of line.
If you know a better way, shoot.